File: selection sort.py

list_of_numbers = []

# swap function take two indexes and swap the elements of those indexes
def swap(a, b):
    c = list_of_numbers[a]
    list_of_numbers[a] = list_of_numbers[b]
    list_of_numbers[b] = c

#selection sort function
def selection_sort():
    n = len(list_of_numbers)
    i = 0
    while i < n:
        j = 0
        max_index = 0
        while j < n-i:
            if list_of_numbers[max_index] < list_of_numbers[j]:
                max_index = j
            j = j + 1
        swap(max_index, n - i-1)  # Swap the maximum number with last element of unsorted subarray
        i = i + 1

#Take list elements from user
def take_input(n):
    for i in range(n):
        list_of_numbers.append(input("Enter number:"))


take_input(int(input("Enter Length of the list:")))
selection_sort()
print("Sorted List:")
print(list_of_numbers)  # print sorted array

File: Selection-Sort.cpp
#include<iostream>

using namespace std;

int list_of_number[5] = { 5, 1,8,2,4 };

void swap(int a, int b) {
    int c = list_of_number[a];
    list_of_number[a] = list_of_number[b];
    list_of_number[b] = c;
}


void selection_sort() {
    int n = sizeof(list_of_number) / sizeof(list_of_number[0]); //length of array
    int i = 0;
    for (int i = 0; i < n; i++) {
        int max_index = 0;
        for (int j = 0; j < n - i; j++) {
            if (list_of_number[max_index] < list_of_number[j]) {
                max_index = j;
            }
        }
        swap(max_index, n - i - 1);
    }
}

void print_array() {
    int n = sizeof(list_of_number) / sizeof(list_of_number[0]); //length of array
    for (int i = 0; i < n; i++) {
        cout << list_of_number[i] << ",";
    }
}

int main() {
    selection_sort();
    print_array();
    return 0;
}